Add more mining variants and fleets

tehhowch opened the pull request:
1. Add mining fleets for Deep, Hai, Unfettered Hai, Korath, Pirates, Remnant, and Wanderers
2. Wanderer's Earth Shaper now is capable of actually mining, and is the primary component of the Wanderer mining fleet.
3. Added new mining ship + mining drone combo, courtesy of @Kryes-Omega 
4. Events which introduce ships will add to those definitions as appropriate
   - Nest, Roost for Pirate, Human mining fleets
   - Mereti miners for Korath Exiles
5. Added fleets to map.txt where appropriate
   - considered normal fleet periods and also the number of minables when setting the mining fleet periods
   - Updated Wanderer events that alter minable systems as necessary to maintain the new mining fleets

Notes:
1. Remnant do not have many places to mine (yet). Not sure how their territory is going to evolve.
2. Included 2 jump-drive capable variants of the Model 8 and a "bodyguard" Model 16 so that Kor Ak'Mari is not mined by just the Raider+Chaser.
3. I did not go through the FW events to check how many Human Miners get removed by the many system fleet respecifications (best done in a different PR) - the answer here is undoubtedly "many / all" for Republic / FW / Syndicate space. I have a PR ready to switch the entirety of Free Worlds story from the static respecifications into `add/remove` format, which will address this persistence.
4. Mining fleet behavior will benefit from #1971 (cargo transfer) and #3064

Bladewood commented on data/korath ships.txt:
```diff
@@ -955,6 +968,23 @@ ship "Model 16"
 	description "The Kor Mereti war drones are constructed in a modular fashion, which allows them to grow over time as more materials become available. This one is recently built, and relatively small."
 
 
+ship "Model 16" "Model 16 (Jump Miner)"
```
"Model 8 (Jump)", "Model 16 (Jump Miner)"?

Also, the Kor Mereti have jump drives?

tehhowch replied:
It has a slicer gun instead of minelayer.

tehhowch replied:
And as implied by other stuff  but not officially stated near the definition, this variant and the jump drive model 8 are only used by the Korath Exiles after they steal the drone tech. Since they have jump drive capability, these and the Sestor drones can too.

Bladewood replied:
I edited my comment after that. My question is, why is Model 8's just "Jump" when 16's is "Jump Miner"? Why are they different?

tehhowch replied:
The model 8's ship definition is only used in mining fleets, so indicating that it is a mining variant isn't really necessary.

tehhowch commented:
@Kryes-Omega has contributed a mining drone and small drone carrier for the Syndicate. I'll let him post the link to the 2x images

Kryes-Omega commented:
The 2x images
![mining carrier 2x](https://raw.githubusercontent.com/Kryes-Omega/2x-ships/master/mining%20carrier%402x.png)
![mining drone 2x](https://raw.githubusercontent.com/Kryes-Omega/2x-ships/master/mining%20drone%20%402x.png)
![plasma drill 2x](https://raw.githubusercontent.com/Kryes-Omega/2x-ships/master/plasma%20drill%402x.png)

Amazinite commented:
Y tho

tehhowch commented:
>Y tho

Why add content to the game? That's a silly question. 
This PR expands the mining fleets used in the game and introduces carried mining ships to both the 'west' and 'east' ends of human space.

Amazinite commented on data/ships.txt:
```diff
@@ -1480,6 +1480,46 @@ ship "Gunboat"
+	explode "tiny explosion" 10
+	explode "small explosion" 10
+	description "The recent asteroid mining boom prompted companies to expand their horizons, and start sending out larger fleets of vessels outfitted for mining. Unfortunately, paying to retrofit cargo ships, as well as crew them is expensive, as there are very few ships that are able to effectively mine."
+	description "	These issues prompted shipyards to start accommodating for miners, but it wasn't enough. So, after petitioning the Republic to allow civilian drone use, the large corporations finally got the Republic parliament to give them rights to use drones, as long as they weren't equipped with any offensive armament."
```
Description of the other ship says that it's built by Syndicated Shipyards, but this makes it sound like multiple shipyards either produce the Harvester or produce similar drones.

Amazinite commented:
Not going to sell the mining vessels or weapon anywhere?

Amazinite commented on data/ships.txt:
```diff
@@ -2136,6 +2176,60 @@ ship "Osprey"
 
 
 
+ship "Prospector"
```
[Lol, why you gotta gype my name when you know you're doing it?](https://github.com/Amazinite/ES-Syndicate-Campaign/blob/master/images/ship/prospector.png)

Kryes-Omega replied:
Only so many words in the english language, man.  I'll look for another name again just to be nice ;)

Amazinite commented:
>This PR expands the mining fleets used in the game and introduces carried mining ships to both the 'west' and 'east' ends of human space.

I don't see the neccessity for the mining ship and drone though.

Amazinite commented on data/weapons.txt:
```diff
@@ -403,6 +403,30 @@ outfit "Plasma Cannon"
+		"firing heat" 1.8
+		"shield damage" .3
+		"hull damage" 3
+	description `The plasma drill is based around the original plasma cannon, scaled down so much though that the range and damage have been greatly reduced. The drill was designed specifically for the Harvester drone, although the demand from solo captains was large enough that the Syndicate released it to be sold individually as well.`
```
A downsized plasma cannon is an interesting choice. My main question would be why? The plasma cannon has the highest shield/hull damage ratio among human weapons (2.52 shield damage for every hull damage). How is it that a downsized plasma cannon has an extremely different damage model (0.1 shield damage for every hull damage). If anything, I think just using a laser would make more sense, or even a downsized proton gun since this *is* being made by the Syndicate.

Amazinite replied:
Gonna take a second to roast the balance of this thing.
*clears throat*

Aside from the strange damage model for a plasma weapon and the fact that it does immense damage for its size (and even more hull DPS than a plasma cannon), this thing produces way too little heat and consumes too little energy. For its current damage model, I'd up the firing energy to at least 0.9, preferable to 1.1. As for the firing heat, I'd up it to at least 3.3 (which even then is generous by my standards).

This thing is CHEAP. I'd at least double the price, but a 50% price hike is good too.

Right now the random velocity from 2 to 4 puts the weapon's range at 150 to 300. For a barrel that long, I think it should be hitting 200 tops. so you could drop the lifetime from 75 to 50, and maybe make its base velocity 3 and random 1 so the range varies from 150 to 200.

Also, no heat damage? Not even a little? It's still a plasma weapon afterall. I'd give it a heat damage of 4.5.

tehhowch replied:
Kryes used a random lifetime, not random velocity, so that's 2 * (75 - 77) not 75 * (2 - 4). I'll let him address the other concerns of yours.

Amazinite replied:
K then, ignore the comments on the range.

Kryes-Omega replied:
as for the first comment, I don't think <insert producer of plasma cannon here> has a patent on plasma based weapon in general? Just as the Syndicate reverse engineered the particle cannon to get their proton gun, I don't think its too much of a stretch. BUT considering @jafdy recent comment, it might be worth reworking the ships to be southbound developed, considering plasma weapons are primarily sold in the south. As for the heat damage, I thought I give it heat damage but it appears to have been lost somewhere in production :P

Amazinite commented on data/ships.txt:
```diff
@@ -1480,6 +1480,46 @@ ship "Gunboat"
+	engine 6.5 37.5
+	engine -6.5 37.5
+	explode "tiny explosion" 10
+	explode "small explosion" 10
```
Why a bigger explosion than a combat drone?

Amazinite commented on data/ships.txt:
```diff
@@ -1480,6 +1480,46 @@ ship "Gunboat"
+	attributes
+		category "Drone"
+		"cost" 100000
+		"shields" 200
```
The military human drones don't have shields. I find it odd that a civilian drone would.

Amazinite commented on data/ships.txt:
```diff
@@ -1480,6 +1480,46 @@ ship "Gunboat"
+	sprite "ship/mining drone"
+	attributes
+		category "Drone"
+		"cost" 100000
```
That's EXPENSIVE for a human drone. I'd put this to at most 80k, just under the combat drone's hull cost. But make sure it stays above 50k, the surveillance drone's hull cost.

Amazinite commented on data/ships.txt:
```diff
@@ -1480,6 +1480,46 @@ ship "Gunboat"
+		"hull" 500
+		"mass" 22
+		"drag" .6
+		"heat dissipation" .92
```
0.92 is greater than both other human drones, each at 0.9. I'd think that this should either match or be beat out by that, dropping maybe to 0.85 if it doesn't match.

Amazinite commented on data/ships.txt:
```diff
@@ -1480,6 +1480,46 @@ ship "Gunboat"
+		"cargo space" 15
+		"outfit space" 56
+		"weapon capacity" 6
+		"engine capacity" 28
```
This is a Syndicate drone, right? Maybe hold off on the release of the drone and then use X1050 Ion Engines (i.e. drop the engine space to 24). We could even change the lore of the X1050s to fit if neccessary, as the only reason they're released alongside the Barb is because of the Barb's low engine capacity; they could just as easily be said to be released for use on the Harvester, and that the Barb is just making use of past inventions.

tehhowch replied:
I know some tweaking of the engine output was needed to make the drone nimble enough to be an effective miner, even with the dedicated turn/thrust engines. The only other route was to change the drag or the mass, which are already pretty limited due to it being a human tech drone.

Amazinite replied:
If that's the case then 28 engine cap is probably fine/

Amazinite commented on data/ships.txt:
```diff
@@ -2136,6 +2176,60 @@ ship "Osprey"
+	sprite "ship/mining carrier"
+	attributes
+		category "Heavy Freighter"
+		"cost" 2340000
```
Exact same hull price as the Hauler II. And judging by some of the other stats, that's what you based it off of. :)
I'd lower this closer to 2 mil, maybe a little lower.

jafdy commented:
I don't think a new ship is needed just to add syndicate mining fleets. I'm sure that there are clever ways of outfitting existing ships to fill that role.
And as Amazinite said the ship sprite doesn't look syndicate. It looks more southbound, and the plasma cannon is a southern weapon as well.

Kryes-Omega commented:
In my opinion, if mining is to be a significant part of the game then there should be equipment that is meant for mining specifically. The larger cargo ships are too slow to catch a lot of the mineables, leaving only light freighters like the clipper or _maybe_ argosy to do mining, or you could have expensive warships do mining with a smaller cargo space. Neither of which seems like something larger mining corporations would want to do, given the option to use mining specific equipment.

As for the look of the ships, like I said to Amazinite you're right they do look more southbound, especially considering my choice of weapon, I suppose reworking them to be southbound might be worth it.

In the mean time, I'll take @Amazinite balancing suggestions and rework the stats for the ships. And i'll use your almighty spreadsheets.

Rob59er commented on data/wanderers.txt:
```diff
@@ -717,6 +737,9 @@ event "wanderers: tempest mass production"
 			"Summer Leaf"
+	fleet "Wanderer Miners"
+		add variant
+			"Tempest"
```
This seems pretty overkill considering all heavier warships would be focused on defensive action and the war effort.

tehhowch replied:
As a low-weight variant (1 in 21 chance to be picked when the `Wanderer Miners` fleet is placed), it was my perception that this would appear as though one of the many Tempests in the system (from the `Wanderer Defense` fleets) was keeping itself busy while patrolling the system. The game's current mechanics will - in the absence of combat targets - rapidly instruct ships to either jump to a new system or land on a planet in the system. This generates traffic volume, but not necessarily a persistent presence (at least until combat starts).

Ideally there would not be any of the dedicated warship designs in the mining fleet, but the Wanderer's stated mining drone has no weapon space and there are no candidate weapons for it to use (e.g. a variant with a baked-in, low-power weapon, like how the Arfecta has a baked-in cloaking device). If someone can make (I don't do artwork well 😁) a lore-compatible mining beam for the Earth Shaper, I would happily rework the `Wanderer Miners` fleet to use it, and feature a rare Summer Leaf, with no Autumn Leaf / Tempests.

tehhowch replied:
Earth Shapers were added in 115e9df84808918adf12af471936c8c29a6b5337, well before mining was added ( f443ada0e1a4c ), so a rework of them to include a mining laser of sorts is plausible.

After I've finished that rework, I'll update the Wanderer Miners fleet to remove their warships. 
Thoughts on keeping the Summer Leaf involved in mining?

Rob59er replied:
some wanderers would want to make some cash off it as well, keep them

tehhowch commented:
Marking as WIP as there are a few outstanding commits that need to arrive before this is ready to merge

tehhowch commented:
I've added my update to the Earth Shaper drone, and reworked the "Wanderer Miners" fleet to contain only the Earth Shaper and the Summer Leaf - the ships developed for the war will only do war stuff :+1: 
 - The `fleeing` personality on the Wanderer Drones fleet ensures that the Earth Shapers spawned there do not attempt to participate in combat (since all `fleeing` does is prevent the ship from targeting a ship, #2658)

Earth Shaper (as seen in the Shipyard)
![image](https://user-images.githubusercontent.com/20871346/29289038-b0bd903c-8100-11e7-8eea-1f4390a261c3.png)

tehhowch commented:
I think content wise this is done, and is ready for merging. I'm going through the FW campaign events as a part of a separate PR that will ensure that fleets defined on map.txt for human systems has a decent shot at actually being present in a new player's game.

Comments welcomed.

endless-sky commented:
I'm a little hesitant to add even more ships and weapons to human space, since there are already so many different ones that nearly every niche is covered.

KiLEdEnNis commented:
I think it's fine here; the Syndicate gets a mining/money-making version of the FW Hauler carriers, and it makes sense lore-wise
it also enables the player to carry 2 of any drone if they want, (some retrofitted drone carrier, even if it'll hardly be used), or for the player to use another drone carrier to hold these mining drones
in short; I think this really adds some new flavor to human space since we lack a dedicated mining ship, and the Syndicate shipyards feel a bit... dull to me personally

Amazinite commented:
@KiLEdEnNis These are Southbound ships, not Syndicate ships.

KiLEdEnNis commented:
I'm sure I read something about the Syndicate... my bad then

tehhowch commented:
One niche not covered is civilian drone and civilian drone carrier. That said, there isn't much use for the player to purchase ships designed for mining until/unless some method of ordering escorts to mine is added. To build off a previous comment, if the player can target an asteroid, then perhaps their carried ships can attack it, as with NPCs and their deployed ships.

endless-sky commented:
Because minables depend on visual identification, and otherwise aren't supposed to be any different from other asteroids, allowing the player to be able to target a minable would require also allowing them to target ordinary asteroids, and that could get rather complicated.

tehhowch commented:
My bit about targeting was in reference to an activated outfit (minable targeting scanner) that requires energy, power, heat management in order to identify minables in a system for a period of time (originally mentioned in a previous pull request). That is understandably a long way off and completely unsupported now, but is something I'm considering putting together for review.

In mods, and potentially the base game, minables are not required to be asteroid lookalikes (e.g. a destroyed ship that can be attacked to yield commodities or a rarely, an outfit).

tehhowch commented:
(I completely agree about the complicated nature of targeting a tiled asteroid)

Kryes-Omega commented:
The Sunflare sprite here doesn't look textured

endless-sky commented:
Mining is intended to be a small part of the game, not a major focus of it. Seeing an NPC ship that is mining should be an unusual event. I feel like adding custom mining ships and weapons and more than a dozen ship variants specifically for mining is overdoing it.

tehhowch commented:
Looks like I've got a plugin to develop then :)
